src/enclave/enclave_event_channel.c

```diff
@@ -168,6 +172,17 @@ void initialize_enclave_event_channel(
     _enc_dev_config = enc_dev_config;
     for (int i = 0; i < evt_channel_num; i++)
     {
+        if (!oe_is_outside_enclave(
```

Comment:
These all look like they're vulnerable to TOCTOU issues.  From line 158, we've established that `enc_dev_config` is outside the enclave, now we're checking the fields while the host can be concurrently modifying them.  Shouldn't we copy this structure into a buffer in the enclave and then check and use the fields?

Reply:
Yes, that's right. According to @prp we can deep-copy all of `enc_dev_config`, so these checks won't be necessary, which is what I'll do now.